### DISCUSSION TITLE
String formatting in .slint

### DISCUSSION THREAD

**Discussion by Blquinn** [slint-ui/slint#3684] [Q&A] [answered]
Hello all,

I was wondering if there's anyway to format a float in .slint markup, to avoid having to do the formatting in my rust code.

I want to do something like `format!("{:.2}", 2.342342525)` in rust. I'd like this in the .slint file because this is how I am displaying the model, so I think it makes sense to go there and not in some additional field on the model.

Follow up question is, can I just insert a snippet of rust code into a slint file somehow?

**Comment by ogoffart** [comment, marked as answer]
Right now you can do:   `"" + (Math.round(2.342342525  * 100) / 100)`

We do need better way to do such formatting, and i think it could be done in a locale definite way through the @tr() macro in the future.

(slightly related: https://github.com/slint-ui/slint/issues/3597  and  https://github.com/slint-ui/slint/discussions/586 )

**Reply by Blquinn** [reply]
Ok, I will go with that approach, thanks!
